examples/tutorial/01_sampling.py

```diff
@@ -1,29 +1,51 @@
-from scannerpy import Database, Job
+from scannerpy import Database, Job, BulkJob
+from scannerpy.stdlib import parsers
 
 ################################################################################
 # This tutorial shows how to select different frames of a video to process.   #
 ################################################################################
 
 with Database() as db:
-
-    # We can access previously created tables with db.table(name).
-    input_table = db.table('example')
+    frame = db.ops.FrameInput()
 
     # You can tell Scanner which frames of the video (or which rows of a video
-    # table) you want to sample. Here, the "strided" sampling mode will run over
-    # every 8th frame, i.e. frames [0, 8, 16, ...]
-    frame = input_table.as_op().strided(8)
+    # table) you want to sample. Here, we indicate that we want to sample
+    # the frame column (we will say how to sample when specifying a job).
+    strided_frame = frame.sample()
+
+    # We process the sampled frame same as before.
+    hist = db.ops.Histogram(frame=strided_frame)
+    output_op = db.ops.Output(columns=[hist])
+
+    # For each job, you can specify how sampling should be performed for
+    # a specific column. In the same way we used the op_args argument to bind
+    # a table to an input column, we bind a sampling directive to strided_frame.
+    job = Job(
+        op_args={
+            frame: db.table('example').column('frame'),
+            # The "strided" sampling mode will run over # every 8th frame,
+            # i.e. frames [0, 8, 16, ...]
+            strided_frame: db.sampler.strided(8),
+            output_op: 'example_hist_strided'
+        }
+    )
+    bulk_job = BulkJob(dag=output_op, jobs=[job])
+    output_tables = db.run(bulk_job, force=True, pipeline_instances_per_node=1)
 
-    # We pass the job to the database same as before, and can process the output
-    # same as before.
-    histogram = db.ops.Histogram(frame = frame)
-    job = Job(columns = [histogram], name = 'example_hist_strided')
-    output_table = db.run(job, force=True)
+    # Loop over the column's rows. Each row is a tuple of the frame number and
+    # value for that row.
+    video_hists = output_tables[0].load(['histogram'], parsers.histograms)
+    num_rows = 0
+    for (frame_index, frame_hists) in video_hists:
+        assert len(frame_hists) == 3
+        assert frame_hists[0].shape[0] == 16
+        num_rows += 1
+    assert num_rows == db.table('example').num_rows() / 8
 
     # Here's some examples of other sampling modes.
-    # Range takes a specific subset of a video. Here, it runs over all frames from
-    # 0 to 100
-    input_table.as_op().range(0, 100)
+    # Range takes a specific subset of a video. Here, it runs over all frames
+    # from 0 to 100
+    db.sampler.range(0, 100)
```

Comment:
General docs question--should we have some kind of API reference, or just have these sorts of functions be documented in small examples via tutorials/applications?

Reply:
I think we should probably have some sort of API reference... maybe something like what flask has? Specifically the API section: http://flask.pocoo.org/docs/0.12/